src/JellyfishClientContext.tsx

```diff
@@ -80,14 +78,13 @@ const JellyfishContextProvider = (props: any) => {
     return () => eventListener.remove();
   }, []);
 
-  const connect = async (url: string, peerToken: string) => {
-    setError(null);
+  const connect = useCallback(async (url: string, peerToken: string) => {
     websocket.current = new WebSocket(url);
 
     const processIncomingMessages = new Promise<void>((resolve, reject) => {
       websocket.current?.addEventListener('close', (event) => {
         if (event.code !== 1000 || event.isTrusted === false) {
-          setError(generateMessage(event));
+          // todo: improve error handling
```

Comment:
This part require further discussion on how to handle such error on both mobile SDK and backedn.